targets/TARGET_STM/TARGET_STM32F4/device/stm32f4xx_hal_pwr.c

```diff
@@ -392,6 +392,9 @@ void HAL_PWR_EnterSLEEPMode(uint32_t Regulator, uint8_t SLEEPEntry)
   {   
     /* Request Wait For Interrupt */
     __WFI();
+    __NOP(); // Workaround for STM32F4 errata
+    __NOP(); // see chapter 2.1.3 - Debugging Sleep/Stop mode with WFE/WFI entry
+    __NOP(); // https://www.st.com/resource/en/errata_sheet/dm00037591-stm32f405-407xx-and-stm32f415-417xx-device-limitations-stmicroelectronics.pdf
```

Comment:
What about the `WFE()` in below lines ? do we need them to be patched as well?

Reply:
Mbed OS doesn't use that mode, so not required for us - would just be an unnecessary image size increase. But if the change is being upstreamed to the HAL, seems like you'd want them there.

Reply:
As the upstream comes to our repo, this will be updated once driver is updated.

@jeromecoutant We fine as it is here now,  correct?

Reply:
I am not sure that ST driver team will accept this change :-(
But as it solves some mbed issues, we are ok to merge this wrkaround.

Reply:
Is there a better alternative, given the errata description? That is the most universally-feasible workaround in your errata sheet.

Reply:
can this be put under DEBUG flag ? this is supposed to happen only when debugging or at least when loading with a DEBUG probe ...  I would prefer RELEASE builds not to be impacted.

Reply:
@LMESTM if you don't mind, we'll proceed with this change for both debug and release for now, as there are other random failures in release mode that we're seeing in CI and need to investigate and rule out for the limited time we have.
Then we can introduce optimizations once we have more information.

Reply:
As soon as confirmed, I'll merge this. It would be good to have this in today.

Reply:
> @LMESTM if you don't mind, we'll proceed with this change for both debug and release for now, as there are other random failures in release mode that we're seeing in CI and need to investigate and rule out for the limited time we have.
> Then we can introduce optimizations once we have more information.

Ok. If RELEASE build is used in CI as well, then the same loading mechanism will apply (thru debug port) and the same issue may arise. So we can't put the proposed compilation switches - please forget my comment and feel free to proceed ...

Reply:
There is a separate argument that something should be done to get that debug/sleep control register back into its default state anyway, otherwise the just-flashed release build in test is not representative of the deployed image in terms of sleep, whether that comes down to power measurements, timing, hitting bugs like this. Arguably release builds should be manually putting that back to its zero state. Flasher should also have taken more care...

Reply:
We think it's the flasher role to do so. If you power cycle the product, the release build will work just fine ...  I'm not sure we should add an explicit reset of those bits, because even the release build should be debugged and in this case the flasher or debugger (or flasher of the debugger) should set the bits.

Reply:
I know bigger systems like Linux are intensely paranoid about chip state post-bootloader, so tend not to rely on any default state. But we can't really afford the ROM space for that level of paranoia in an embedded image.

And yes, in this specific case, I see your point that a debugger may obviously have deliberately  set those bits before running us, for a good reason, so an image should not reset any bits there, regardless of the general issue.